cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -866,70 +657,54 @@ def bulk_migrate_from_modulestore(
 
         # Forwarding legacy content to migrated content
         status.set_state(MigrationStep.FORWARDING.value)
-        if forward_source_to_target:
-            for source_data in source_data_list:
-                if not source_data.migration.is_failed:
-                    _forwarding_content(source_data)
+        for source_data in source_data_list:
+            if forward_source_to_target is False:
+                continue  # Explicitly requested not to forward.
+            if forward_source_to_target is None and not source_data.previous_migration:
+                # Unspecified whether or not to forward.
+                # So, forward iff there was no previous migration.
+                continue
+            if source_data.migration.is_failed:
+                # Don't forward failed migrations.
+                continue
+            _forward_content(source_data)
```

Comment:
We've expanded this logic to allow for `forward_source_to_target is None`, i.e., "let the system decide whether to forward source->target, based on whether there's been a previous successful migration".